src/commands/publish/mod.rs

```diff
@@ -177,9 +178,22 @@ fn make_public_on_subdomain(project: &Project, user: &GlobalUser) -> Result<(),
     Ok(())
 }
 
-fn build_js_script() -> Result<String, failure::Error> {
+fn build_js_script() -> Result<Form, failure::Error> {
     let package = Package::new("./")?;
-    Ok(fs::read_to_string(package.main()?)?)
+    let script_path = package.main()?;
+    let metadata_json = r#"{"body_part":"script","bindings":[]}"#;
+
+    let metadata = Part::text(metadata_json)
+        .file_name("metadata.json")
+        .mime_str("application/json")?;
+
+    let form = Form::new()
+        .file("script", script_path)
+        .unwrap_or_else(|_| panic!("{} not found. Did you rename your js files?", script_path));
+
+    form.part("metadata", metadata);
+
+    Ok(form)
 }
 
 fn build_multipart_script() -> Result<Form, failure::Error> {
```

Comment:
can we make this work for webpack + rust project types?

Reply:
probably yes. the subject of this pr is to consistently use multipart upload. subsequent PRs will focus on unifying the way the multipart form is constructed to be more flexible.